### PR TITLE
eth_getBalance

### PR DESCRIPTION
**Pull request by DockBoss**

### CODE REVIEW

File: json-rpc/eth_getBalance.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# `Eth_getBalance`
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in [RFC-2119](https://www.ietf.org/rfc/rfc2119.txt).
+Specification | Description 
+---|---
+1|Eth_getBalance must be sent a valid Ethereum address as the first parameter|
```
Comment (tkstanczak) [inline json-rpc/eth_getBalance.md:5]: I think this is not true.

Comment (tkstanczak) [reply json-rpc/eth_getBalance.md:5]: If you mean formatting, this should be covered in a metadocument for the address type

File: json-rpc/eth_getBalance.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+1.1 |Eth_getBalance should return error code -32602 "invalid argument 0: json: cannot unmarshal hex string of odd length into Go value of type common.Address" if the address send has an odd length|
```
Comment (tkstanczak) [inline json-rpc/eth_getBalance.md:6]: interestingly this message looks very bad and seems to be an internal Geth error so a better message should be suggested

File: json-rpc/eth_getBalance.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+1.2 |Eth_getBalance should return error code -32602 "invalid argument 0: hex string has length n, want 40 for common.Address" if address length is even but not equal to 40|
+1.3 |Eth_getBalance should return error code -32602  "invalid argument 0: json: cannot unmarshal hex string without 0x prefix into Go value of type common.Address" if the address is blank or a random string|
+2 |Eth_getBalnace must be sent the block that you want the balance from as the second parameter|
```
Comment (tkstanczak) [inline json-rpc/eth_getBalance.md:9]: this is very inefficient way of describing JSON RPC, it seems. Parameters and return values should be described in the EIP-1474 format while edge cases as a list of edge cases

File: json-rpc/eth_getBalance.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+2.1|Eth_getBalance must accept block numbers as hex string "0xa" block 10, "0xf5f" block 3935|
```
Comment (tkstanczak) [inline json-rpc/eth_getBalance.md:10]: This should be covered in the meta info about the types.

File: json-rpc/eth_getBalance.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+2.1.1 |Eth_getBalance should return error code -32602 "invalid argument 1: hex string \"0x\""  if it receives an incomplete  hex string|
```
Comment (tkstanczak) [inline json-rpc/eth_getBalance.md:11]: Maybe all errors should be defined in a separate section for errors? In a nice table.

File: json-rpc/eth_getBalance.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+2.1.2 |Eth_getBalance should return  error code -32000 "header not found" if block does not exist|
+2.2 |Eth_getBalance must also accept the tags "latest" "earliest" "pending"|
```
Comment (tkstanczak) [inline json-rpc/eth_getBalance.md:13]: should be covered by main types

File: json-rpc/eth_getBalance.md [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+2.2.1 |Eth_getBalance should return error code -32602 "invalid argument 1: hex string without 0x prefix" if it receives a string that is not "latest", "earliest", or "pending"|
+3 |Eth_getBalance must return the balance of the account in wei as type quantity|
```
Comment (tkstanczak) [inline json-rpc/eth_getBalance.md:15]: return value should be described next to parameters in the main description of the endpoint

**Comment by SamWilsn** [comment]
Should be migrated to https://github.com/ethereum/eth1.0-apis

**Pull request closed without merging**